plugins/widget/plugin.js

```diff
@@ -2701,6 +2733,24 @@
 				var attrs = element.attributes,
 					widget, widgetElement;
 
+				// Reset initial and trailing space by replacing &nbsp; with white space (#605).
+				if ( 'data-cke-widget-white-space' in attrs ) {
```

Comment:
Same here, a more generic solution is needed. Maybe when replacing spaces you could tag an element in which space was replaced with something similar to `data-cke-widget-white-space` so then it is easier to find it in this code (assuming that there can be many nested elements)? Additionally, you may also mark widgets (same as now) so you don't have to iterate over all of them.